core/arch/arm/mm/core_mmu_v7.c

```diff
@@ -687,16 +697,43 @@ static void map_memarea(struct tee_mmap_region *mm, uint32_t *ttb)
 		if ((mm->va | mm->pa | mm->size) & SMALL_PAGE_MASK)
 			panic("memarea can't be mapped");
 
+		attr = mattr_to_desc(2, mm->attr);
+		if (attr == INVALID_DESC)
```

Comment:
I'd prefer `if (!(mm->attr & TEE_MATTR_VALID_BLOCK))` as this retrieved value of `attr` isn't used.